### PR TITLE
The limit was used as a string instead of as an integer.

### LINKED ISSUE #33: limit parameter in Sampler incorrectly parsed
From: https://hpcgitlab.inl.gov/idaholab/raven/issues/683

The 'limit' parameter in the Sampler base class is parsed as a string rather than an int, which makes the count check in Sampler's amIreadyToProvideAnInput erroneously always return True even if the limit has been exceeded.

Changing this line in readSamplerInit(): self.limit = childChild.text to: self.limit = int(childChild.text) fixes the problem.

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
This converts it to an integer.
Closes #33

--------
Pull Requeset Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
#33 

##### What are the significant changes in functionality due to this change request?
The limit is an integer.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/Samplers/Sampler.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def readSamplerInit(self,xmlNode):
         self.initSeed = Distributions.randomIntegers(0,2**31,self)
         for childChild in child:
           if childChild.tag == "limit":
-            self.limit = childChild.text
+            self.limit = int(childChild.text)
```
Comment (joshua-cogliati-inl) [inline framework/Samplers/Sampler.py]: Possibly this should be in a try catch ValueError block (like Samplers.MonteCarlo does) to give a better error.

Comment (PaulTalbot-INL) [inline framework/Samplers/Sampler.py]: I wholeheartedly agree with that idea.  Would it be too painful to add that as part of this MR?

**Comment by joshua-cogliati-inl** [comment]
Hm, https://github.com/idaholab/raven/pull/103 removes this code altogether.

**Comment by alfoa** [comment]
Is this MR addressed in #103 ?

**Comment by alfoa** [comment]
Since it is addressed in MR #103 , Would be ok to close this?

**Comment by joshua-cogliati-inl** [comment]
Yes, it is okay to close this since it should be addressed in https://github.com/idaholab/raven/pull/103

**Pull request closed without merging**